drivers/adc/adc_stm32.c

```diff
@@ -267,6 +269,10 @@ static int start_read(struct device *dev, const struct adc_sequence *sequence)
 #endif
 	data->channel_count = 1;
 
+#if defined(CONFIG_SOC_SERIES_STM32WBX)
```

Comment:
Out of curiosity, why is this required for STM32WB only (and not similar SoC L4 for instance)?
This would require a comment to make it clear

Reply:
It looks like it isn't actually needed since the ADC defaults to single diff.  I'll remove it.